### DISCUSSION TITLE
Injected className from db does not seem to work for responsive padding

### DISCUSSION THREAD

**Discussion by timmeade** [tailwindlabs/tailwindcss#6105] [Help] [answered]
OK odd one for me...  React NextJS 11 latest (not beta) tailwind

I am injecting adding additional className based on element from database.  

When adding md:pt-3 it does nothing either with hot reload or full reload.  Changing md:text-2xl to md:text-3xl DOES work for both hot reloading and reloading.   To add to weirdness;  removing size breakpoint 'pt-3' and it works fine.  

Also;  if I add md:pt-3 to the actual className and NOT the db section, it does work and still works when removed if that (md:pt-3) is added to db add section.  That is until changed (md:pt-2) and it disappears completely. 

The breakpoint and db section does appear in elements tab.  but nothing changes on screen.

How i am injecting it.  Originally it was using the classNames function from tailwindui and that was exactly same result.

```
    <div
      className={`flex flex-row items-center gap-2 ${
        profileLine.format ? profileLine.format : ''
      } `}
```

In the following everything injected works except the md:pt-4. if I change to injecting just pt-4 it will work.   If I add the md:pt-4 to the static portion it works there so I know it's not being overwritten or something.  Odd......

```
<div class="flex flex-row items-center gap-2 text-lg md:text-2xl md:pt-4 ">Title Here</div>
```

I did a little more testing and the weirdness increased:

Injecting

```
text-lg md:text-xl pt-2 md:pt-2
```
Show pt-2 on screen

```
text-lg md:text-2xl pt-2 md:pt-2
```
text changes on md to 2xl

```
text-lg md:text-2xl pt-2 md:pt-4
```
nothing changes anywhere except looking in Elements I do see the md:pt-4 but nothing changes on screen

```
text-lg md:text-2xl pt-4 md:pt-2
```
padding top changes to -4 everywhere

**Comment by RobinMalfait** [comment, marked as answer]
Hey! Thank you for your bug report!
Much appreciated! 🙏

You have to keep in mind that Tailwind is a **build**-time tool.If you inject classes from a database at runtime, then Tailwind doesn't know about it because we don't ship a runtime.
When you say that certain classes don't work, double check your devtools that you see actual css for that class, which it probably doesn't.

The reason I think it works for those other classes is probably because they already exist _somewhere_ in your codebase. Which means that we already generated the css for it and therefore you will see some results on your screen.

Long story short, make sure to [safelist](https://tailwindcss.com/docs/optimizing-for-production#safelisting-specific-classes) all the classes that you want to generate if they are coming from a database. This will ensure that the classes that will come from the db at some point int time will already exist in your final css file.

If you are allowing to generate _any_ tailwind class, then we have a bigger problem (because we don't ship a runtime). In that case you probably want to create a node API (or something similar) where tailwind generates css files on demand based on the contents of the database. But this will require you to build some infrastructure and will require some effort to make it work smoothly and caches properly.

**Reply by timmeade** [reply]
And that was the perfect answer.  Much thanks.   Changed purge to:

```
  purge: {
    content: [
      './pages/**/*.{js,ts,jsx,tsx}',
      './components/**/*.{js,ts,jsx,tsx}',
      './containers/**/*.{js,ts,jsx,tsx}',
    ],
    safelist: ['md:pt-2', 'md:pt-3', 'md:pt-4'],
  },
```

Works as anticipated.   Onwards.
